internal/k8shandler/log-exploration-api.go

```diff
@@ -0,0 +1,180 @@
+package k8shandler
+
+import (
+	route "github.com/openshift/api/route/v1"
+	"github.com/openshift/cluster-logging-operator/internal/factory"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	logAPIPort = 8080
+)
+
+func newLogExplorationAPIRoute(routeName, namespace, serviceName, componentName, loggingComponent string) *route.Route {
+	return &route.Route{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Route",
+			APIVersion: route.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      routeName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"component":     componentName,
+				"logging-infra": loggingComponent,
+			},
+		},
+		Spec: route.RouteSpec{
+			To: route.RouteTargetReference{
+				Name: serviceName,
+				Kind: "Service",
+			},
+		},
+	}
+}
+
+func (clusterRequest *ClusterLoggingRequest) createLogExplorationAPIService() error {
+	desired := factory.NewService(
+		"logexplorationapi-service",
+		clusterRequest.Cluster.Namespace,
+		"logexplorationapi",
+		[]v1.ServicePort{
+			{
+				Port:       logAPIPort,
+				TargetPort: intstr.FromInt(logAPIPort),
+			},
+		},
+	)
+
+	utils.AddOwnerRefToObject(desired, utils.AsOwner(clusterRequest.Cluster))
+	err := clusterRequest.Create(desired)
+	return err
+}
+func (clusterRequest *ClusterLoggingRequest) createLogExplorationAPIRoute() error {
+	apiRoute := newLogExplorationAPIRoute("logexplorationapi-route", clusterRequest.Cluster.Namespace, "logexplorationapi-service", "logexplorationapi", "logexplorationapi")
+	utils.AddOwnerRefToObject(apiRoute, utils.AsOwner(clusterRequest.Cluster))
+	err := clusterRequest.Create(apiRoute)
+	return err
+}
+func (clusterRequest *ClusterLoggingRequest) createLogExplorationAPIDeployment() error {
+
+	logApiPodSpec := newLogExplorationApiPodSpec()
+
+	logApiDeployment := NewDeployment("logexplorationapi", clusterRequest.Cluster.Namespace, "logexplorationapi", "logexplorationapi", logApiPodSpec)
```

Comment:
Make use of constants.  Refactor strings into constants

Reply:
Renamed service using a constant in pull request #1208.